lib_eio/workpool.mli

```diff
@@ -0,0 +1,37 @@
+type t
+
+(** Creates a new workpool with [domain_count].
+
+    [domain_concurrency] is the maximum number of jobs that each domain can run at a time.
+
+    [transient] (default: true). When true, the workpool will not block the [~sw] Switch from completing.
+      When false, you must call [terminate] to release the [~sw] Switch. *)
+val create :
+  sw:Switch.t ->
+  domain_count:int ->
+  domain_concurrency:int ->
+  ?transient:bool ->
+  #Domain_manager.t ->
+  t
+
+(** Run a job on this workpool. It is placed at the end of the queue. *)
+val run : t -> (unit -> 'a) -> ('a, exn) result
+
+(** Same as [run] but raises if the job failed. *)
+val run_exn : t -> (unit -> 'a) -> 'a
+
+(** Same as [run] but returns immediately, without blocking. *)
+val run_promise : sw:Switch.t -> t -> (unit -> 'a) -> ('a, exn) result Promise.t
+
+(** Waits for all running jobs to complete, then returns.
+    No new jobs are started, even if they were already enqueued.
+    To abort all running jobs instead of waiting for them, call [Switch.fail] on the Switch used to create this workpool *)
+val terminate : sw:Switch.t -> t -> unit
```

Comment:
Why is there a `sw` argument here?

Reply:
Good catch, I'm not sure how I missed that.